common/src/IO/WorldReader.cpp

```diff
@@ -101,11 +105,14 @@ namespace TrenchBroom {
             // handle default layer attributes, which are stored in worldspawn
             auto* defaultLayerNode = m_world->defaultLayer();
             for (const Model::EntityAttribute& attribute : attributes) {
-                if (attribute.name() == Model::AttributeNames::LayerColor
-                    || attribute.name() == Model::AttributeNames::LayerOmitFromExport) {
-                    auto defaultLayerEntity = defaultLayerNode->entity();
-                    defaultLayerEntity.addOrUpdateAttribute(attribute.name(), attribute.value());
-                    defaultLayerNode->setEntity(std::move(defaultLayerEntity));
+                if (attribute.name() == Model::AttributeNames::LayerColor && Color::canParse(attribute.value())) {
+                    auto defaultLayer = defaultLayerNode->layer();
+                    defaultLayer.setColor(Color::parse(attribute.value()));
+                    defaultLayerNode->setLayer(std::move(defaultLayer));
+                } else if (attribute.name() == Model::AttributeNames::LayerOmitFromExport) {
+                    auto defaultLayer = defaultLayerNode->layer();
+                    defaultLayer.setOmitFromExport(true);
```

Comment:
This should match the logic in MapReader.cpp which checks for `value == Model::AttributeValues::LayerOmitFromExportValue`, rather than just setting `setOmitFromExport(true);` unconditionally.

Unfortunately with the current setup, we have the parsing logic duplicated for regular layers + the default layer

Reply:
44a82a427